job.js

```diff
@@ -0,0 +1,192 @@
+        const jobOpts = this.getQueryOptions({ dryRun: true });
+
+        try {
+            const [job] = await this.bigQuery.createQueryJob(jobOpts);
+            const cost = _getCost(job.metadata.statistics.totalBytesProcessed);
+
+            if (cost.gb >= this.costThresholdInGB) {
+                this.logger.notify('BigQuery Job | Expensive Query').steps(0,1).msg('bigquery-job.js Expensive query over threshold. name:%s costThresholdInGB:%s. Estimated cost: $%s | %s TB | %s GB | %s KB | %s MB | %s bytes', this.name, this.costThresholdInGB, cost.price, cost.tb, cost.gb, cost.mb, cost.kb, cost.bytes)
+            }
+            else {
+                this.logger.info('bigquery-job.js Validated job. name:%s. Estimated cost: $%s | %s TB | %s GB | %s KB | %s MB | %s bytes', this.name, cost.price, cost.tb, cost.gb, cost.mb, cost.kb, cost.bytes)
+            }
+        }
+        catch(err) {
+            BigQueryError.parseErrorAndThrow(err);
+        }
+    }
+
+    /**
+     * @return {Promise<Object[]>}
+     */
+    async run() {
+        let elapsed = Duration.start();
+        await this.validate();
+
+        this.logger.info('bigquery-job.js Running job. name:%s', this.name);
+        const jobOpts = this.getQueryOptions({ dryRun: false });
+
+        try {
+            const [job] = await this.bigQuery.createQueryJob(jobOpts);
+            this.logger.info('bigquery-job.js Executed job. Getting query results. name:%s', this.name);
+
+            const [response] = await job.getMetadata();
+            const cacheHit = response.statistics.query.cacheHit;
+            const cost = _getCost(response.statistics.query.totalBytesBilled);
+            this.logger.info('bigquery-job.js Start getting results. name:%s costThresholdInGB:%s cacheHit:%s. Billed cost: $%s | %s TB | %s GB | %s KB | %s MB | %s bytes', this.name, this.costThresholdInGB, cacheHit, cost.price, cost.tb, cost.gb, cost.mb, cost.kb, cost.bytes);
+
+            const [rows] = await this._getQueryResults(job);
+            this.logger.info('bigquery-job.js Got query results. name:%s costThresholdInGB:%s cacheHit:%s totalRows:%s elapsed:%s ms. Billed cost: $%s | %s TB | %s GB | %s KB | %s MB | %s bytes', this.name, this.costThresholdInGB, cacheHit, rows.length, elapsed.end(), cost.price, cost.tb, cost.gb, cost.mb, cost.kb, cost.bytes);
+
+            return rows;
+        }
+        catch(err) {
+            BigQueryError.parseErrorAndThrow(err);
+        }
+    }
+
+    /**
+     * @param job
+     * @return {Promise<QueryRowsResponse> | void}
+     * @private
+     */
+    _getQueryResults(job) {
+        return job.getQueryResults({ autoPaginate: true });
+    }
+}
+
+/**
+ * @param {String} bytes
+ * @return {{mb: string, bytes: string, price: string, gb: string, tb: string}}
+ * @private
+ */
+function _getCost(bytes) {
+    let kb = parseInt(bytes) / 1024;
+    let mb = kb / 1024;
+    let gb = mb / 1024;
+    let tb = gb / 1024;
+    let price = tb * 5;
```

Comment:
The value of the terabyte should be configurable. If Google change the price per terabyte, this cost will be wrong.